tempMonApp/src/tempMon.c

```diff
@@ -405,3 +410,57 @@ cpu_thermal_init(int hwmon_inst, struct temp_state* state)
   state->fd = open(file, O_RDONLY);
   return state->fd < 0 ? -1 : 0;
 }
+
+/******************************************************************************
+ * IOC shell function registration
+ *****************************************************************************/
+
+static void
+hwmonStatusCallFunc(const iocshArgBuf* args)
+{
+  /** Caller wants us to re-index all hwmon types */
+  if (args[0].ival != 0) {
+    printf("Available hwmon sensors:\n");
+
+    /** Loop through all hwmon types and read their type */
+    for (int i = 0; i < 99; ++i) {
+      char buf[128];
+      snprintf(buf, sizeof(buf), "/sys/class/hwmon/hwmon%d/name", i);
+      int fd = open(buf, O_RDONLY);
+      if (fd < 0)
+        break;
+
+      char name[128];
+      ssize_t l = 0;
+      if ((l = read(fd, name, sizeof(name)-1)) <= 0) {
+        close(fd);
+        continue;
+      }
+      name[l] = 0;
+      close(fd);
+
+      printf(" %s", name);
+    }
+    return;
+  }
+
+  /** Otherwise just report on what's being used */
+  printf("hwmon type: %s\n", hwmon_type);
+}
+
+/** This is defined as a "registrar" in the dbd file. It will be called when
+  * registerRecordDeviceDriver is, before iocBoot.
+  */ 
+void
+tempMonRegister()
+{
+  /* hwmonStatus */
+  {
+    static const iocshArg arg0 = {"bReIndex", iocshArgInt};
```

Comment:
Now I see that it re-indexes only when the bReIndex is used on the status function. It makes sense to teach about IOCshell arguments, but this idea got me confused.

Reply:
Yeah I wanted to demonstrate arguments here. A simple example function is probably better than this :)